sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -315,13 +317,31 @@ def scheduled_enqueue_time_utc(self, value):
 
     @property
     def body(self):
-        # type: () -> Optional[Union[bytes, Iterable[bytes]]]
-        """The body of the Message.
-
-        :rtype: bytes or Iterable[bytes]
+        # type: () -> Optional[Union[bytes, Iterable[bytes], List, Iterable[List], Any]]
+        """The body of the Message. The format may vary depending
+        on the body type:
+        For ~azure.servicebus.AMQPMessageBodyType.DATA, the body could be bytes or Iterable[bytes]
+        For ~azure.servicebus.AMQPMessageBodyType.SEQUENCE, the body could be List or Iterable[List]
+        For ~azure.servicebus.AMQPMessageBodyType.VALUE, the body could be any type.
+
+        :rtype: Optional[Union[bytes, Iterable[bytes], List, Iterable[List], Any]]
```

Comment:
will this generate properly in docstring?
afaik, it should be 
```suggestion
        :rtype: Optional[Union[bytes, Iterable[bytes], List, Iterable[List], Any]]
        :rtype: bytes or Iterable[bytes] or list or Iterable[list]
```

also, why everything else if there's Any in rtype

Reply:
hmm I do wonder if we should consider this breaking....
Maybe not? I mean, if the sender started sending a different body type that the receiver wasn't prepared for - the receiving code would likely be broken anyway....

Reply:
this is a good question as we're expanding the returning types.

I agree, it would be broken anyway... -- I'm thinking of other options like raising error if body is not of data body type and introducing new apis for getting sequence and value, but still being breaking, users have to update their receiving part to handle different body types.

torn on this one 😣

let's keep the design for now and iterate over it in next sprint? or do you think it's risky to have this shipped?

Reply:
i'm fine with iterating for the next sprint

Reply:
updated the return type to be Any and created an issue to keep the discussion: https://github.com/Azure/azure-sdk-for-python/issues/17786